ingest/oni/dns_master.py

```diff
@@ -3,107 +3,127 @@
+        self._queue_name = conf['queue_name']
+
+        self._s3_bucket = conf['s3_bucket']
+        self._staging_folder = conf['staging_folder']
+        self._archive_folder = conf['archive_folder']
+
+
+    def start(self):
+
+        try:
+            while True:
+                print "Watching the S3 Bucket: {0} to collect files".format(self._s3_bucket)
+                file_list = client.list_objects(Bucket=self._s3_bucket, Prefix=self._staging_folder)['Contents']
+                for key in file_list:
+                    file_name_parts = key['Key'].split('/')
+                    file_name = file_name_parts[len(file_name_parts) - 1]
+                    filename_str = str(file_name)
+                    # get file name and date
+                    if filename_str and (not filename_str.isspace()):
+                        print "New file: {0}".format(file_name)
+                        binary_year, binary_month, binary_day, binary_hour, binary_date_path = Util.build_hdfs_path(file_name, self._dsource)
+                        # local storage path
+                        file_local_path = '../stage/{0}'.format(file_name)
+                        #aws path with timestamp.
+                        aws_archive_path = "{0}/binary/{1}/{2}".format(self._archive_folder, binary_date_path, binary_hour)
+                        # get timestamp from the file name.
+                        file_date = file_name.split('.')[0]
+                        pcap_hour = file_date[-6:-4]
+                        pcap_date_path = file_date[-14:-6]
+                        # get file from AWS_s3
+                        client.download_file(self._s3_bucket, '{0}/{1}'.format(self._staging_folder, file_name), file_local_path)
+                        # move binary file to archive for holding
+                        s3.Object(self._s3_bucket, '{0}/{1}'.format(aws_archive_path, file_name)).copy_from(CopySource='{0}/{1}/{2}'.format(self._s3_bucket, self._staging_folder, file_name))
+                        # delete staging file in s3
+                        s3.Object(self._s3_bucket, '{0}/{1}'.format(self._staging_folder, file_name)).delete()
+                        print "file :{0} staged".format(file_name)
+                        if file_name.endswith('.pcap'):
+                            self._process_pcap_file(file_name, file_local_path)
+                            print "file :{0} processed".format(file_name)
+                time.sleep(self._time_to_wait)
+
+        except KeyboardInterrupt:
+            print "exiting"
+
+    def _process_pcap_file(self, file_name, file_local_path):
+
+        local_path = os.getenv('LUSER')
+
+        # get file size.
+        file_size = os.stat(file_local_path)
+        if file_size.st_size > 1145498644:
+
+            # split file.
+            self._split_pcap_file(file_name, file_local_path, local_path)
+        else:
+
+            # send rabbitmq notification.
```

Comment:
this comment is redundant with the method name below (or there is an important distinction about notifying RabbitMQ that is not suggested in the method call)